android/src/main/java/io/invertase/firebase/links/RNFirebaseLinks.java

```diff
@@ -0,0 +1,294 @@
+import android.util.Log;
+
+import java.util.Map;
+
+import com.facebook.react.bridge.ActivityEventListener;
+import com.facebook.react.bridge.LifecycleEventListener;
+import com.facebook.react.bridge.Promise;
+import com.facebook.react.bridge.ReactApplicationContext;
+import com.facebook.react.bridge.ReactContextBaseJavaModule;
+import com.facebook.react.bridge.ReactMethod;
+import com.facebook.react.bridge.ReadableMap;
+
+import com.google.firebase.dynamiclinks.DynamicLink;
+import com.google.firebase.dynamiclinks.FirebaseDynamicLinks;
+import com.google.firebase.dynamiclinks.ShortDynamicLink;
+import com.google.firebase.dynamiclinks.PendingDynamicLinkData;
+
+import com.google.android.gms.tasks.Task;
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.OnSuccessListener;
+import com.google.android.gms.tasks.OnFailureListener;
+
+import io.invertase.firebase.Utils;
+
+public class RNFirebaseLinks extends ReactContextBaseJavaModule implements ActivityEventListener, LifecycleEventListener {
+  private final static String TAG = RNFirebaseLinks.class.getCanonicalName();
+  private String mInitialLink = null;
+  private boolean mInitialLinkInitialized = false;
+
+  private interface ResolveHandler {
+    void onResolved(String url);
+  }
+
+  private interface  ErrorHandler {
+    void onError(Exception e);
+  }
+
+  public RNFirebaseLinks(ReactApplicationContext reactContext) {
+    super(reactContext);
+    getReactApplicationContext().addActivityEventListener(this);
+    getReactApplicationContext().addLifecycleEventListener(this);
+  }
+
+  @Override
+  public String getName() {
+    return "RNFirebaseLinks";
+  }
+
+
+  private void resloveLink(Intent intent, final ResolveHandler resolveHandler, final ErrorHandler errorHandler) {
+    FirebaseDynamicLinks.getInstance()
+      .getDynamicLink(intent)
+      .addOnSuccessListener(new OnSuccessListener<PendingDynamicLinkData>() {
+        @Override
+        public void onSuccess(PendingDynamicLinkData pendingDynamicLinkData) {
+          if (pendingDynamicLinkData != null) {
+            Uri deepLinkUri = pendingDynamicLinkData.getLink();
+            String url = deepLinkUri.toString();
+            resolveHandler.onResolved(url);
+          } else {
+            resolveHandler.onResolved(null);
+          }
+        }
+      })
+      .addOnFailureListener(new OnFailureListener() {
+        @Override
+        public void onFailure(@NonNull Exception e) {
+          errorHandler.onError(e);
+        }
+      });
+  }
+
+  @ReactMethod
+  public void getInitialLink(final Promise promise) {
+    if (mInitialLinkInitialized) {
+      promise.resolve(mInitialLink);
+    } else {
+      Activity activity = getCurrentActivity();
+      if (activity != null) {
+        resloveLink(activity.getIntent(), new ResolveHandler() {
+          @Override
+          public void onResolved(String url) {
+            if (url != null) {
```

Comment:
It was done because of this: https://firebase.google.com/docs/reference/android/com/google/firebase/dynamiclinks/FirebaseDynamicLinks.html#getDynamicLink(android.content.Intent)

The getDynamicLink method usually takes the url from the server. Therefore in the second time getInitialLink invoked, getDynamicLink will return null. In order to be consistent to the iOS implementation where this value is kept in userActivityDicionary I store the initial link in a member, and cleans it on host destroy lifecycle method.